### PR TITLE
feat: Theme Context, Theme Provider & Theme Toggle

### PR DESCRIPTION
**Pull request by horsemaker**
- Added Theme Context
- Added Theme Provider
- Added Theme Toggle

Preview:

recording

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  --color-background-accent: #3b3b3b;
+  --color-foreground: #ffffff;
+  --color-note-1: #ffab91;
+  --color-note-2: #ffcc80;
+  --color-note-3: #e6ee9b;
+  --color-note-4: #80deea;
+  --color-note-5: #cf93d9;
+  --color-note-6: #f48fb1;
+  --color-note-7: #80cbc4;
+}
```
Comment (t007rushi) [inline src/App.css:29]: Liked that you note color for both the themes

**Review by narharikale — APPROVED** [review]
clean code and awesome transition 🤩

File: src/frontend/contexts/theme-context.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+} from "react";
+import { KAFKA_THEME } from "../constants";
+
+const ThemeContext = createContext();
+
+const preferColorSchemeQuery = "(prefers-color-scheme: dark)";
+
+const giveInitialTheme = () =>
+  localStorage.getItem(KAFKA_THEME) ||
+  (matchMedia(preferColorSchemeQuery).matches ? "dark" : "light");
```
Comment (t007rushi) [inline src/frontend/contexts/theme-context.js:16]: Nicely done for browser theme clever approach

**Review by t007rushi — APPROVED** [review]
Theme slider animation is 🔥

**Review by adidoshi — APPROVED** [review]
Great implementation of dark theme mode.
- Nice use of `useLayoutEffect`
- Wow, first time heard of `matchMedia`
- Toggle switch works smoothly, great work

**Pull request merged**